cmake/compilerDefinitions.cmake

```diff
@@ -37,9 +37,13 @@ if(HAVE_RULES)
     add_definitions(-DHAVE_RULES)
 endif()
 
+if(USE_INT128)
+    add_definitions(-DHAVE_INT128)
+endif()
+
 if(Boost_FOUND)
     add_definitions(-DHAVE_BOOST)
-    if(USE_BOOST_INT128)
+    if(USE_INT128 AND USE_BOOST_INT128)
         add_definitions(-DHAVE_BOOST_INT128)
     endif()
```

Comment:
I think the logic around `USE_BOOST_INT128` is flawed . `On` is supposed to *force* the usage of Boost and `Auto` should provide a fallback to Boost if no built-in type exists but Boost is enabled (i.e. Visual Studio).

Reply:
hmm.. how about:
```
VALUEFLOW_INT128=no
VALUEFLOW_INT128=gcc
VALUEFLOW_INT128=boost
```
?